### DISCUSSION TITLE
Filtering a relationship selection based on another field

### DISCUSSION THREAD

**Discussion by PhilBookst** [keystonejs/keystone#7467] [Feature requests]
Hey Keystone Team,

I'm currently struggling with a functionality where I need to create a dynamic amount of questions with multiple possible answers and then connect a question to an article and pick one of the answers.

Basically I need to create a question e.g.: "When do you wake up?"
And then provide multiple answers e.g.: "At 6AM", "Before 12PM", "After 12PM" 

In the article I would then pick the question "When do you wake up?" 
and the answers I can pick from should be one of "At 6AM", "Before 12PM", "After 12PM" 

I got this partially working with this schema:
```ts
export const Question = list({
  fields: {
    question: text({
      validation: {
        isRequired: true
      }
    }),
    answer: relationship({
      ref: "Answer.question",
      many: true,
    }),
  },
  ui: { labelField: "question" }
})

export const Answer = list({
  fields: {
    answer: text(),
    question: relationship({
      ref: "Question.answer",
      many: false
    })
  },
  ui: { labelField: "answer" },
})

```

This allows me to create multiple answers to a specific question. 
Now I need to connect these question to a article, and within there I want to select a question and then select on of the possible answers to it.

I've tried creating another list like this: 
```ts
export const QuestionList = list({
  fields: {
    question: relationship({
      ref: "Question",
    }),
    answer: relationship({
      ref: "Answer",
    }),
  },
})

```

which I then link to a article via another relationship field on the article list with the many flag. 
Then on the article ItemView I pick a question and need to have my answers filtered based on the question I picked.

Is this possible somehow?

**Comment by gwyneplaine** [comment]
@PhilBookst thanks for this, presently this is not something that Keystone supports, I've changed this to a *feature request* instead, and we've started conversations internally around this use-case.

**Comment by fkrauthan** [comment]
I have a similar use-case: For example a submission in my case belongs to a team who is part of a course. So if I now select in the admin UI a course that submission belong to I want to only see teams in the team selector that are part of that course.

**Comment by fatihsolhan** [comment]
Haven't tested again but this workaround should work
https://github.com/keystonejs/keystone/discussions/7424#discussioncomment-2491522

**Reply by fkrauthan** [reply]
Ah so just creating my own component todo the filtering? Yeah I still hope this can be implemented soon as it seems like a pretty common use-case.

**Comment by TheRealFlyingCoder** [comment]
Just wanted to drop my use case in here as well as this feature is very useful for staff workflows.

In my eCommerce platform, I have Brands, Brand Colours, and Products. Brand Colours are necessary because "White" from one brand can be a completely different colour to "White" from another brand, but all "White" products inside one brand are the same.

Upon Product creation, a Brand will be selected from the List. <-- The current state of the form now has "Brand"

Next, the user has to choose which Brand Colour's this product can come in, Right now the dropdown will include all Brand Colours for all the brands, resulting in "WHITE" "WHITE" "WHITE" for example which is a problem.

Ideally given the form state knows which Brand we have, it should be able to filter the results, so that it only shows BrandColours that have the selected Brand relationship.

TLDR: **Essentially this feature request is for adding a filter function to a relationship's query, which passes along the current state of the List being editted, along with the results of the query.**
